entity-framework/core/testing/index.md

```diff
@@ -1,117 +1,26 @@
+description: Overview of testing applications that use Entity Framework Core
+author: roji
+ms.date: 01/17/2021
 uid: core/testing/index
 ---
-# Testing code that uses EF Core
+# Testing EF Core Applications
 
-Testing code that accesses a database requires either:
+Testing is an important concern to almost all application types - it allows you to be sure your application works correctly, and makes it instantly known if its behavior regresses in the future. Since testing may affect how your code is architected, it's highly recommended to plan for a testing early and to ensure good coverage as your application evolves. This introductory section provides a quick overview of various testing strategies for applications using EF Core.
 
-* Running queries and updates against the same database system used in production.
-* Running queries and updates against some other easier to manage database system.
-* Using test doubles or some other mechanism to avoid using a database at all.
+## Involving the database (or not)
 
-This document outlines the trade-offs involved in each of these choices and shows how EF Core can be used with each approach.
+When writing tests for your EF Core application, one basic decision you need to make is whether your tests will involve your production database system - just as your application does - or whether your tests will run against a [test double](https://martinfowler.com/bliki/TestDouble.html), which replaces your production database system. Two prominent examples of test doubles in the EF Core context are [SQLite in-memory mode](xref:core/testing/choosing-a-testing-strategy#sqlite-as-a-database-fake), and the [InMemory provider](xref:core/testing/choosing-a-testing-strategy#inmemory-as-a-database-fake). For simplicity's sake, we'll use the term *integration tests* to refer to tests which involve your production database systems, and *unit tests* to refer to tests which use a test double.
 
-> [!TIP]
-> See [EF Core testing sample](xref:core/testing/testing-sample) for code demonstrating the concepts introduced here.
+For an in-depth comparison and analysis of integration and unit testing approaches in EF Core applications, see [Choosing a testing strategy](xref:core/testing/choosing-a-testing-strategy). Below is a short point-by-point summary to help you get up to speed with the different options:
 
-## All database providers are not equal
+* Developers frequently turn to unit testing because they believe testing against their production database system is difficult or slow. This isn't always true in our experience, and we suggest giving this approach a chance: [Integration testing](xref:core/testing/integration-testing) provides techniques for reliable, fast integration testing. Writing at least some integration tests is usually necessary in any case - to make sure your application actually works against your production database - and unit testing can be limited in what it allow you to test (see below).
```

Comment:
```suggestion
* Developers frequently turn to unit testing because they believe testing against their production database system is difficult or slow. This isn't always true in our experience, and we suggest giving this approach a chance: [Integration testing](xref:core/testing/integration-testing) provides techniques for reliable, fast integration testing. Writing at least some integration tests is usually necessary in any case - to make sure your application actually works against your production database - and unit testing can be limited in what it allows you to test (see below).
```